File: MyNotesProject/CommonLibrary/src/main/java/com/fsk/common/database/DatabaseStorable.java
package com.fsk.common.database;


import android.database.sqlite.SQLiteDatabase;

/**
 * An interface that defines the storable values.
 */
public interface DatabaseStorable {

    /**
     * Store the value in the supplied database.
     *
     * @param db
     *         the database to receive the value.
     */
    void save(SQLiteDatabase db);

    /**
     * Remove the value from the database.
     *
     * @param db
     *         the database to delete the value from.
     */
    void delete(SQLiteDatabase db);
}

File: MyNotesProject/CommonLibrary/src/main/java/com/fsk/common/database/DatabaseStorable.kt
package com.fsk.common.database

import android.database.sqlite.SQLiteDatabase

/**
 * An interface that defines the Database storable values.
 */
interface DatabaseStorable {

    /**
     * Store the value in the supplied database.
     *
     * @param db
     *          the database to receive the value.
     */
    fun save(db: SQLiteDatabase);

    /**
     * Remove the value from the database.
     *
     * @param db
     *           the database to delete the value from.
     */
    fun delete(db: SQLiteDatabase);
}